src/main/java/org/broadinstitute/hellbender/utils/read/CigarUtils.java

```diff
@@ -106,7 +106,7 @@ public static int countRefBasesBasedOnCigar(final GATKRead read, final int cigar
         for(int i = cigarStartIndex; i < cigarEndIndex; i++){
             final CigarElement cigarElement = elems.get(i);
             final CigarOperator operator = cigarElement.getOperator();
-            if (operator.consumesReferenceBases() || operator == CigarOperator.S) {
+            if (operator.consumesReferenceBases() || operator.isClipping()) {
                 result += cigarElement.getLength();
```

Comment:
I see that given the original code, this is the wright translation so no changes needed... but the documentation is rather unclear about what this method does. Perhaps a better name for it would be ```countReferenceBasesOnUnclippedAlignment```.

My point here is that now we have a some methods that count soft and hard-clips in this kind of operations whereas some other methods only du soft... and I wonder whether there is a need for consolidate these (e.g. perhaps some changes in so calling code would make such a difference unnecessary). 

In any case feel free to leave it for a future PR.